src/classes/Contacts.cls

```diff
@@ -0,0 +1,748 @@
+    }
+
+    private static String namingExclusionsFor(Contact con) {
+        return con.npo02__Naming_Exclusions__c;
+    }
+
+    /*******************************************************************************************************
+    * @description Copies the values from the naming exclusions checkboxes to the multiselect picklist.
+    * @param con The Contact
+    * @return void
+    ********************************************************************************************************/
+    public void copyNamingExclusionsFromCheckboxes() {
+        for (SObject sobj : getRecords()) {
+            Contact con = (Contact) sobj;
+            copyNamingExclusionsFromCheckboxes(con);
+        }
+    }
+
+    private void copyNamingExclusionsFromCheckboxes(Contact con) {
+        NamingExclusions exclusions = buildNamingExclusionsFromCheckboxesFor(con);
+        con.npo02__Naming_Exclusions__c = exclusions.getDisplayValue();
+    }
+
+    private NamingExclusions buildNamingExclusionsFromCheckboxesFor(Contact con) {
+        NamingExclusions exclusions = new NamingExclusions(null);
+        if (con.Exclude_from_Household_Name__c) {
+            exclusions.excludeName();
+        }
+        if (con.Exclude_from_Household_Formal_Greeting__c) {
+            exclusions.excludeFormalGreeting();
+        }
+        if (con.Exclude_from_Household_Informal_Greeting__c) {
+            exclusions.excludeInformalGreeting();
+        }
+        return exclusions;
+    }
+
+    /*******************************************************************************************************
+    * @description Creates a new Household Object for each contact, unless excluded by recordType
+    * @param contacts List of Contacts
+    * @param isInsertTrigger Whether called from the afterInsert trigger (vs. afterUpdate trigger) on Contacts
+    * @return void
+    ********************************************************************************************************/
+    private void insertHouseholdsForContacts(List<Contact> contacts, Boolean isInsertTrigger) {
+        List<npo02__Household__c> householdInserts = getHouseholdsForInsert(contacts);
+        if (householdInserts.size() > 0) {
+            Database.SaveResult[] householdSaveResults =
+                    UTIL_DMLService.insertRecords(householdInserts, false);
+
+            List<Contact> contactsToBeUpdatedWithHouseholdId =
+                    processHouseholdSaveResults(contacts, householdSaveResults);
+            update contactsToBeUpdatedWithHouseholdId;
+
+            if (isInsertTrigger) {
+                updateNamesAfterHouseholdInsert(householdSaveResults);
+            }
+        }
+    }
+
+    private List<Contact> processHouseholdSaveResults(List<Contact> contacts,
+            Database.SaveResult[] householdSaveResults) {
+        List<Contact> contactsToBeUpdatedWithHouseholdId = new List<Contact>();
+        Integer i = 0;
```

Comment:
This variable is not used and can be removed.

Reply:
Tried removing but looks like it IS actually used!  Ha - I didn't see that either - used to grab the correct result a few lines below this.